### ISSUE TITLE
Mungo should respect existing MIME types

### ISSUE THREAD

**Issue by mamurphy** [omniti-labs/mungo#1]
**Reported by jschulz on 1 Jan 1970 00:19 UTC**
Mungo currently defaults to serving all files as text/html.  It should try to use any existing Apache (mod_mime) settings before finally falling back to text/html.

'''To reproduce''', add the following to httpd.conf

```
<Files ~ (\.(asp|css|js|html|xml))>    
   SetHandler  perl-script
   PerlSetVar StatINC 1
   PerlHandler Mungo
</Files>

```

Now, all files with those extensions will be served as text/html, even JavaScript and CSS.
If a doctype of:

```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
                                "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```

is added, the page will "break" because the CSS and JavaScript will not be recognized as the appropriate type of file.

The page will lack any CSS styling and any JavaScript will not function.

'''Remedy:'''
You can add the following to httpd.conf

```
<Files ~ (\.(css))> 
   PerlSetVar MungoContentType 'text/css'
</Files>

<Files ~ (\.(js))> 
   PerlSetVar MungoContentType 'text/javascript'
</Files>

<Files ~ (\.(xml))> 
   PerlSetVar MungoContentType 'text/xml'
</Files>

```

but Mungo should respect (or at least try to obtain) those types already set in conf/mime.types.

If reading from conf/mime.types is too difficult/expensive, a single place (mungo.mime ?) to record all those associations would be appreciated.

**Comment by mamurphy** [comment]
**Comment by jesus on 1 Jan 1970 00:19 UTC**
(In [5]) fixes #1

**Comment by mamurphy** [comment]
**Modified by clinton on 1 Jan 1970 00:20 UTC**
